### PR TITLE
Table view

### LINKED ISSUE #1063: User Interface Modification - Add an option to show projects as a list... SP 4
### Discussed in https://github.com/openkfw/TruBudget/discussions/1040

<div type='discussions-op-text'>

<sup>Originally posted by **LukasKM** January 28, 2022</sup>
When there is a higher number of projects, the current "card view" becomes difficult to navigate in, despite the existing search function. 
It would be great if users have an option to switch from "card view" to a "list view" that would allow to show more projects on the screen. 

This could be combined with: 

1. Filtering of closed projects (e.g. simle switch include/exclude closed projects).
2. Sorting of projects according to the project status, i.e. the open projects on the top, closed on the bottom.
3. Grouping of projects (in a list view) by their tags, e.g. if there are several projects with the same tags, they would be shown within the same goup in that list.</div>

### PR DESCRIPTION
**Pull request by daniel-arnauer**
### Description
Add a Data Table for projects

- [x] Pagination
- [x] Sorting
- [x] Filtering / Searching
- [ ] e2e-tests

Closes #1063

### CODE REVIEW

File: frontend/src/helper.js [hunk 1]
```diff
@@ -102,10 +102,15 @@ export const validateLanguagePattern = amount => {
 export const numberSignsRegex = /^[0-9,.-]*$/;
 
 export const unixTsToString = ts => {
-  let dateString = dayjs.unix(ts).format("MMM D, YYYY");
+  let dateString = dayjs.unix(ts).format("D MMM YYYY");
   return dateString;
 };
```
Comment (daniel-arnauer) [inline frontend/src/helper.js:107]: This format can be exported directly in csv and I think it looks better

**Comment by daniel-arnauer** [comment]
To Do:

- [ ] Add View-link
- [ ] Add Buttons (view, edit, permission)
- [ ] Check if CSV format is needed
- [ ] Add time to date
- [ ] Rebuild the filter component: Per default, show search bar. open an accordion to show Datepicker, Radio Buttons for Status, DropDown for Assignees
- [ ] Implement Filter component on Project, subproject, workflowitem level

**Comment by daniel-arnauer** [comment]
More Feedback:
- [ ] Add Tags
- [ ] Remove Download button since excel-export already exists
- [ ] Add filter to suproject, workflowitems

File: frontend/src/pages/Common/NewDropdown.js [hunk 2]
```diff
@@ -78,6 +79,12 @@ const Dropdown = props => {
                 />
               </div>
             </div>
+            {/* TODO refactor to this:
```
Comment (mayrmartin) [inline frontend/src/pages/Common/NewDropdown.js:82]: remove todo?

File: frontend/src/helper.js [hunk 3]
```diff
@@ -102,10 +102,15 @@ export const validateLanguagePattern = amount => {
 export const numberSignsRegex = /^[0-9,.-]*$/;
 
 export const unixTsToString = ts => {
-  let dateString = dayjs.unix(ts).format("MMM D, YYYY");
+  let dateString = dayjs.unix(ts).format("D MMM YYYY HH:mm");
```
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply frontend/src/helper.js:105]: MMM means the month is written as word, not as number

**Review by mayrmartin — APPROVED** [review]

**Pull request merged**